tests/components/vodafone_station/test_coordinator.py

```diff
@@ -0,0 +1,216 @@
+        assert state.state == STATE_UNAVAILABLE
+
+
+async def test_coordinator_uptime_shift(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory
+) -> None:
+    """Test Device cleanup on coordinator update."""
+
+    entry = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    entry.add_to_hass(hass)
+
+    with (
+        patch("aiovodafone.api.VodafoneStationSercommApi.login") as mock_login,
+        patch(
+            "aiovodafone.api.VodafoneStationSercommApi.get_devices_data",
+            return_value=DEVICE_DATA_QUERY,
+        ) as mock_devices_data,
+        patch(
+            "aiovodafone.api.VodafoneStationSercommApi.get_sensor_data",
+            return_value=SENSOR_DATA_QUERY,
+        ) as mock_sensor_data,
+        patch(
+            "datetime.datetime.now",
+            return_value=datetime.strptime(
+                "2024-12-02T13:00:00+0000", "%Y-%m-%dT%H:%M:%S%z"
+            ),
+        ),
+    ):
+        uptime = "2024-11-19T20:19:00+00:00"
+        uptime_entity = (
+            f"sensor.vodafone_station_{SENSOR_DATA_QUERY['sys_serial_number']}_uptime"
+        )
+
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        mock_login.assert_called_once()
+        mock_devices_data.assert_called_once()
+        mock_sensor_data.assert_called_once()
+
+        state = hass.states.get(uptime_entity)
+        assert state
+        assert state.state == uptime
+
+        mock_sensor_data.reset_mock()
+
+        SENSOR_DATA_QUERY["sys_uptime"] = "12:17:23"
+        mock_sensor_data.return_value = SENSOR_DATA_QUERY
+
+        freezer.tick(SCAN_INTERVAL)
+        async_fire_time_changed(hass)
+        await hass.async_block_till_done(wait_background_tasks=True)
+
+        state = hass.states.get(uptime_entity)
+        assert state
+        assert state.state == uptime
+
+
+async def test_coordinator_device_cleanup(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory
+) -> None:
+    """Test Device cleanup on coordinator update."""
+
+    entry = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    entry.add_to_hass(hass)
+
+    with (
+        patch("aiovodafone.api.VodafoneStationSercommApi.login") as mock_login,
+        patch(
+            "aiovodafone.api.VodafoneStationSercommApi.get_devices_data",
+            return_value=DEVICE_DATA_QUERY,
+        ) as mock_devices_data,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        mock_login.assert_called_once()
+        mock_devices_data.assert_called_once()
+
+        mock_devices_data.reset_mock()
+        mock_devices_data.return_value = DEVICE_2
+
+        freezer.tick(SCAN_INTERVAL)
+        async_fire_time_changed(hass)
+        await hass.async_block_till_done(wait_background_tasks=True)
+
+        state = hass.states.get(f"device_tracker.vodafone_station_{DEVICE_1_MAC}")
+        assert state is None
```

Comment:
So what is this testing? If we expect device 1 to be there before we skip time, let's assert that first